### DISCUSSION TITLE
Add optional text field for panel link target name

### DISCUSSION THREAD

**Discussion by adeverteuil** [grafana/grafana#38346] [Dashboards, Panels & Visualizations]
**What would you like to be added**:

The current form has a **Open in new tab** toggle switch.

![image001](https://user-images.githubusercontent.com/3270755/64279052-33680600-cf1c-11e9-9ca0-74345b1b6c0c.jpg)

This adds the HTML attribute `target="_blank"`.

We would like the ability to specify a name in a text field for the target attribute value.

`target="_blank"` → Each click opens in a new tab.  
`target="foo"` → First click opens in a new tab. Subsequent clicks opens in the existing named context "foo".

Reference: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a

**Why is this needed**:

Could be used by Editors to customize the workflow for Viewers in a business where clicking panel links all day results in an unmanageable number of browser tabs.

In previous versions of Grafana, we were able to inject Javascript to do this.

![image002](https://user-images.githubusercontent.com/3270755/64278855-bd639f00-cf1b-11e9-8152-df72c4094c60.jpg)

Note the link URL field contains this string, starting with the closing quote character:

```
" onclick=" this.setAttribute('target','Instance');  this.setAttribute('href', '/d/zcYld2iZk/business-process-instance-main-view?var-BPInstanceID=${__cell}' );
```

**Comment by iroks** [comment]
any news to this request?

I would strongly support this feauture. Our current workaround is to provide a custom link at 
.../master/grafana/ContextMenu.tsx:
in the line:
target="myFrame" //instead of "{item.target}"

if there is such frame available on the page, .e.g. in a text panel (
<iframe allowtransparency="true" style="background: #000000;"  src="" name="myFrame" width="0" height="0" frameBorder="0" scrolling="no" ></iframe>) 

than the context will be shown in this frame; if not a new window will be opened. Potential security feautre: link in this frame can load a Javascript that can manipulate the entire document (this feauture should remain available). 

Therefore, the simplest solution could be to provide a custom target name instead of _blank that can be set in the configuration (custom for all types of links). The better solution is to provide a custom name for every link (much better,  .../grafana/DataLinkEditor.tsx?).

Why is this needed:
we are using Grafana to broadcast monitoring (audio/video) and a click to a data link enables to listen the audio or see the video in the embedded player to chosen date in the same gui environment.

Thanks in advance for your feedback.

**Comment by iroks** [comment]
just checked the latest Grafana version; the above mentioned proposal will not work due to the source code changes. It seems that _blank is still hard coded on multiple locations in the source code in Grafana 8.*. Therefore, the simplest solution is still to provide a custom target name (that could be configured statically in the configuration) instead of _blank. If such custom target not exist in the document, the browser opens new tab. Security is an issue that should be discusses/addressed.

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
